Example/Example Swift/ComponentAPIDocumentsService.swift

```diff
@@ -121,148 +109,93 @@ final class ComponentAPIDocumentsService: ComponentAPIDocumentServiceProtocol {
+                                       metadata: metadata) { result in
+                                        switch result {
+                                        case .success(let createdDocument):
+                                            print("📄 Created document with id: \(createdDocument.id) " +
+                                                "for vision document \(document.id)")
+                                            completion(.success(createdDocument))
+                                        case .failure(let error):
+                                            print("❌ Document creation failed")
+                                            
+                                            completion(.failure(error))
+                                        }
+                                        
+        }
     }
     
-    fileprivate func deleteCompositeDocument(withId id: String) {
-        giniSDK.sessionManager
-            .getSession()
-            .continueWith(block: sessionBlock(cancellationToken: nil))
-            .continueOnSuccessWith(block: { [weak self] _ in
-                self?.giniSDK.documentTaskManager.deleteCompositeDocument(withId: id,
-                                                                          cancellationToken: nil)
-            })
-            .continueWith(block: { task in
-                if task.isCancelled || task.error != nil {
-                    print("❌ Error deleting composite document with id: \(id)")
-                } else {
-                    print("🗑 Deleted composite document with id: \(id)")
-                }
-                
-                return nil
-            })
-        
+    func delete(composite document: Document) {
+        documentService.delete(document) { result in
+            switch result {
+            case .success:
+                print("🗑 Deleted composite document with id: \(document.id)")
+            case .failure:
+                print("❌ Error deleting composite document with id: \(document.id)")
+            }
+        }
     }
     
-    fileprivate func deletePartialDocument(withId id: String) {
-        giniSDK.sessionManager
-            .getSession()
-            .continueWith(block: sessionBlock(cancellationToken: nil))
-            .continueOnSuccessWith(block: { [weak self] _ in
-                self?.giniSDK.documentTaskManager.deletePartialDocument(withId: id,
-                                                                        cancellationToken: nil)
-            })
-            .continueWith(block: { task in
-                if task.isCancelled || task.error != nil {
-                    print("❌ Error deleting partial document with id: \(id)")
-                } else {
-                    print("🗑 Deleted partial document with id: \(id)")
-                }
-                
-                return nil
-            })
-        
+    func delete(partial document: Document) {
```

Comment:
The two delete methods are nearly identical. I'd merge them (even if it means that the logs don't specify if it was a partial or composite)

Reply:
Makes sense. I was only doing it for the logs, but I can print the `sourceClassification` instead